### PR TITLE
Code excerpter

### PR DESCRIPTION
**Pull request by dmdashenkov**
This PR integrates the repository with the [`code_excerpter`](https://github.com/chalin/code_excerpter) tool for maintaining the relevancy of code examples.

The simple in-place code examples are easy to create but hard to maintain. Especially when changes in framework break the example code.

For the purpose of maintaining the example code adequate, we should put it all in a Gradle project which will be built on CI in pull requests. This new Gradle project is located in `_examples/examples` directory. The `check-examples` GitHub action builds the project to make sure it compiles.

`code_excerpter` is a tool targeting primarily dart projects. Thus, the `_examples` directory is now a Dart project.

## Updating snippets

To update code snippets, run `./_script/excerpt`.
The script requires the `pub` tool to be installed and available in the `$PATH`.

## Adding new snippets

To add a new code snippet, add the following construct to the doc:
```
<?code-excerpt "path/to/file (doc-region)"?>
` ` `
` ` `
```

That is, a `<?code-excerpt?>` tag followed by a code fence (with the right language). The code fence
may be empty, since it will be automatically overwritten.

The path to file should be relevant to the `_examples/examples`. The doc region should match the name of a doc region in the specified file.

See `_examples/README.md` for more details.

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

File: _examples/examples/config/gradle/dependencies.gradle [hunk 1]
```diff
@@ -0,0 +1,292 @@
+/*
+ * Copyright 2020, TeamDev. All rights reserved.
```
Comment (armiol) [inline _examples/examples/config/gradle/dependencies.gradle:2]: I don't understand whether this file is in this repository, or it is grabbed from the `config` submodule, as it should.
Please review.

The same applies to the rest of the files in this folder.

File: _examples/examples/config/version.gradle [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * project-specific general dependencies.
+ *
+ * This file is used in both module `build.gradle` scripts and in the integration tests,
+ * as we want to manage the versions in a single source.
+ * 
+ * This file is copied to the root of the project ONLY if there's no file with such a name
+ * already in the root directory.
+ */
+
+final def SPINE_VERSION = '1.0.1'
```
Comment (armiol) [inline _examples/examples/config/version.gradle:32]: And this is really weird.

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

**Comment by alexander-yevsyukov** [comment]
@dmdashenkov, please also add `README.md` files under `_samples` and `_examples` which explain the purpose of the directories and rules for adding content to them.

File: _examples/README.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+// #enddocregion region-name
+
+Here goes more irrelevant code.
+```
+
+Only the lines between `#docregion` and `#enddocregion` are included into the snippet.
+
+### Joining pieces
+
+Doc regions can be joined from different pieces along the file. For that, just declare several dec
```
Comment (alexander-yevsyukov) [inline _examples/README.md:65]: > `several dec`
Dod you mean “several doc”?

File: _examples/examples/build.gradle [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+plugins {
+    id "io.spine.tools.gradle.bootstrap" version "1.5.0"
```
Comment (alexander-yevsyukov) [inline _examples/examples/build.gradle:22]: 1.5.8 is already available. Let's use it. The newer version gives more concise API for testing.

**Comment by dmdashenkov** [comment]
@alexander-yevsyukov, PTAL.

File: _samples/README.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+When writing an entry to the site, its often handy to use code snippets to demo the described
+concepts. Those code snippets should be kept up to date with the framework.
+
+For this purpose, we integrate with the [`code_excerpter`](https://github.com/chalin/code_excerpter)
+and [`code_excerpt_updater`](https://github.com/chalin/code_excerpt_updater) tools.
+
+Note that this directory should only contain small examples. For substantial examples involvoing
+multiple concepts interacting with each other, see `_examples`.  
+
+Since the tools are designed for Dart projects, the contents of `_samples` directory are shaped as
```
Comment (alexander-yevsyukov) [inline _samples/README.md:12]: How about saying this: 

> ... the directory `_samples` is shaped as a Dart project where the example code must be placed under the `examples` dir. The `examples` dir, in turn, contains example source code in Java, and as such is shaped as a Gradle project.


...

**Pull request merged**